src/V3UniqueNames.h

```diff
@@ -28,14 +28,24 @@
 #include <unordered_map>
 
 class V3UniqueNames final {
+public:
+    enum runMode {
+        NORMAL,
+        NO_SUFFIX  // V3UniqueNames will not add a suffix to the name if it is unique.
+    };
+
+private:
     const std::string m_prefix;  // Prefix to attach to all names
 
     std::unordered_map<std::string, unsigned> m_multiplicity;  // Suffix number for given key
 
```

Comment:
std::map.  The unordered map will cause perf problem given where you are now using it in emit more often.

Reply:
Fixed